### DISCUSSION TITLE
Permission denied error when running git-aggregate on a new project

### DISCUSSION THREAD

**Discussion by mtelahun** [Tecnativa/doodba-copier-template#507] [Q&A] [answered]
Hello everyone,

I have been running Doodba for years on my laptop without any problems, but I recently got a new laptop with Ubuntu 22.04 and tried to run doodba on it. The steps that I followed are:

mtm@storm:~/code/aq-roses-bishoftu-doodba$ id
uid=1000(mtm) gid=1000(mtm) groups=1000(mtm),4(adm),24(cdrom),27(sudo),30(dip),46(plugdev),100(users),114(lpadmin),127(libvirt),984(docker)

copier copy --trust --vcs-ref=v7.0.0 gh:Tecnativa/doodba-copier-template aq-roses-bishoftu-doodba
cd aq-roses-bishoftu-doodba
invoke develop
invoke img-build --pull
invoke git-aggregate

And at this point I get an immediate Permission denied error:

mtm@storm:~/code/aq-roses-bishoftu-doodba$ invoke git-aggregate
Reinitialized existing Git repository in /home/mtm/code/aq-roses-bishoftu-doodba/.git/
pre-commit installed at .git/hooks/pre-commit
time="2024-11-19T17:44:34+03:00" level=warning msg="/home/mtm/code/aq-roses-bishoftu-doodba/setup-devel.yaml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion"
doodba INFO: Running gitaggregate with /opt/odoo/custom/src/repos.yaml
(INFO) [14:44:36] git_aggregator.repo  odoo   Start aggregation of /opt/odoo/custom/src/odoo
(INFO) [14:44:36] git_aggregator.repo  odoo   Init empty git repository in /opt/odoo/custom/src/odoo
(INFO) [14:44:36] git_aggregator.repo  trevi-addons   Start aggregation of /opt/odoo/custom/src/trevi-addons
(INFO) [14:44:36] git_aggregator.repo  trevi-addons   Init empty git repository in /opt/odoo/custom/src/trevi-addons
fatal: cannot mkdir /opt/odoo/custom/src/odoo: Permission denied
fatal: cannot mkdir /opt/odoo/custom/src/trevi-addons: Permission denied
(ERROR) [14:44:36] git_aggregator.repo  odoo   /opt/odoo/custom/src/odoo> error calling ['git', 'init', '/opt/odoo/custom/src/odoo']
(ERROR) [14:44:36] git_aggregator.repo  trevi-addons   /opt/odoo/custom/src/trevi-addons> error calling ['git', 'init', '/opt/odoo/custom/src/trevi-addons']

I seem to vaguely remember encountering this before when I first started using doodba (in the early  years: 201x) and I obviously solved it, but I can't remember how I did it. Can anyone here give me a hint? Thanks.

**Comment by chienandalu** [comment]
Have you set proper docker permissions?

**Reply by mtelahun** [reply]
Yes, I am in the docker group and the docker socket under ~/.docker has proper permissions

**Comment by mtelahun** [comment, marked as answer]
This seems related to docker -desktop. I switched to running system-wide docker (as root) and the problem no longer exists.
